components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/LocalEntryResource.java

```diff
@@ -0,0 +1,179 @@
+
+        String entryKey = Utils.getQueryParameter(messageContext, Constants.NAME);
+        if (Objects.nonNull(entryKey)) {
+            populateLocalEntryData(axis2MessageContext, synapseConfiguration, entryKey);
+        } else {
+            populateLocalEntries(axis2MessageContext, synapseConfiguration);
+        }
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        return true;
+    }
+
+    /**
+     * Sets the list of all available local entries to the response as json.
+     *
+     * @param axis2MessageContext axis2 message context
+     * @param synapseConfiguration Synapse configuration object
+     */
+    private void populateLocalEntries(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                      SynapseConfiguration synapseConfiguration) {
+
+        Map definedEntries = synapseConfiguration.getDefinedEntries();
+        // (-2) to account for the 2 local entries defined by the server
+        JSONObject jsonBody = Utils.createJSONList(definedEntries.size() - 2);
+        for (Object o : definedEntries.entrySet()) {
+            Map.Entry entry = (Map.Entry) o;
+            addLocalEntryToJsonList(entry, jsonBody.getJSONArray(Constants.LIST));
+        }
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+    }
+
+    /**
+     * Adds the provided Entry to the json array provided.
+     * @param list json array
+     * @param entry Map.Entry
+     * */
+    private void addLocalEntryToJsonList(Map.Entry entry, JSONArray list) {
+
+        // Escape local entries defined by the server
+        String key = (String) entry.getKey();
+        if (SynapseConstants.SERVER_IP.equals(key)
+                || SynapseConstants.SERVER_HOST.equals(key)) {
+            return;
+        }
+
+        if (entry.getValue() instanceof Entry) {
+            Entry localEntry = (Entry) entry.getValue();
+            String entryType = getEntryType(localEntry);
+            JSONObject entryObject = new JSONObject();
+            entryObject.put(Constants.NAME, key);
+            entryObject.put(TYPE_ATTRIBUTE, entryType);
+            list.put(entryObject);
+        }
+    }
+
+    /**
+     * Returns the type of the specified local entry.
+     * @param localEntry local entry
+     * @return entry type String
+     * */
+    private String getEntryType(Entry localEntry) {
+
+        String entryType;
+        switch (localEntry.getType()) {
+            case Entry.REMOTE_ENTRY:
+                entryType = "Registry Key";
+                break;
+            case Entry.INLINE_TEXT:
+                entryType = "Inline Text";
+                break;
+            case Entry.INLINE_XML:
+                entryType = "Inline XML";
+                break;
+            case Entry.URL_SRC:
+                entryType = "Source URL";
+                break;
+            default:
+                entryType = "Unknown";
```

Comment:
Would it be a better idea to may be set the entry type to "Unknown type: " +localEntry.getType() ".